platform-dependent-components/problem-solver/cxx/identifiers-module/agent/translate_main_system_idtfs_from_sc_to_file_agent.cpp

```diff
@@ -73,76 +68,79 @@ SC_AGENT_IMPLEMENTATION(TranslateMainSystemIdtfsFromScToFileAgent)
   bool const & resultOfWrite = WriteInFile(strIdtfs);
 
   if (resultOfWrite)
+  {
     SC_LOG_DEBUG("File has been created");
+    return action.FinishSuccessfully();
+  }
   else
+  {
     SC_LOG_ERROR("File hasn't been created");
-  utils::AgentUtils::finishAgentWork(&m_memoryCtx, actionAddr, resultOfWrite);
-  SC_LOG_DEBUG("TranslateMainSystemIdtfsFromScToFileAgent finished");
-  return SC_RESULT_OK;
+    return action.FinishUnsuccessfully();
+  }
+}
+
+ScAddr TranslateMainSystemIdtfsFromScToFileAgent::GetActionClass() const
+{
+  return IdentifiersKeynodes::action_find_identifiers;
 }
 
 bool TranslateMainSystemIdtfsFromScToFileAgent::CheckAction(ScAddr const & actionAddr)
```

Comment:
not used method

Reply:
tests are failed without it 
![image](https://github.com/user-attachments/assets/ec626757-5476-4976-b41b-2a61749b119e)